application/libraries/REST_Controller.php

```diff
@@ -1028,21 +1028,28 @@ protected function _detect_lang()
      */
     protected function _log_request($authorized = FALSE)
     {
+        
+        $this->rest->db->select_max("ID","insert_id"); //Gets last ID because not all databases support IDENTITY columns (Like Oracle)
+        $insert_id = $this->rest->db->get($this->config->item('rest_logs_table'));
+        $insert_id = $insert_id->row(); //gets the result array
+        $insert_id = (count($insert_id)>0) ? $insert_id->insert_id+1 : 1; //if there's no rows in the table there'll be a null result then we set the first id to be 1, otherwise it'll be the greatest one plus 1
```

Comment:
Don't forget to add spaces where applicable. +1 is a little hard to read and isn't conforming to the CI3 style guide.